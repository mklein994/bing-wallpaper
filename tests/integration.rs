macro_rules! project {
    ($dir:literal) => {
        concat!(
            env!("CARGO_MANIFEST_DIR"),
            "/tests/local/",
            $dir,
            "/bing_wallpaper"
        )
    };

    ($dir:literal, $file:literal) => {
        concat!(project!($dir), "/", $file)
    };
}

const OPT_ARGS: &[&str] = &[
    "--config-path",
    project!("config", "config.json"),
    "--data-path",
    project!("share"),
    "--state-path",
    project!("state", "image_index.json"),
];

fn get_output<I, S>(args: I) -> (String, String)
where
    I: IntoIterator<Item = S>,
    S: AsRef<std::ffi::OsStr>,
{
    use std::process::Command;

    let output = Command::new(env!("CARGO_BIN_EXE_bing-wallpaper"))
        .args(OPT_ARGS)
        .args(args)
        .output()
        .unwrap();

    let stdout = String::from_utf8(output.stdout).unwrap();
    let stderr = String::from_utf8(output.stderr).unwrap();

    (stdout, stderr)
}

#[test]
fn list_images() {
    let (stdout, stderr) = get_output(["list-images"]);
    insta::assert_snapshot!(stdout);
    insta::assert_snapshot!(stderr);
}

#[test]
fn end_to_end_test() {
    let (stdout, stderr) = get_output(["project-dirs"]);
    insta::with_settings!({filters => vec![
    (env!("CARGO_MANIFEST_DIR"), ""),
    ]}, {
        insta::assert_snapshot!(stdout);
        insta::assert_snapshot!(stderr);
    });
}
