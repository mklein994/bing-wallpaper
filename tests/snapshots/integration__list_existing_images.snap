---
source: tests/integration.rs
expression: stdout
---
Game on	fcd58e5358a8b390cb537e4075a8df36_OHR.ParalympicsParis_EN-CA3661228731_UHD.jpg
A Catalan cliffhanger	2304d6b4f07a54942d753b8a87d00d97_OHR.CastellfollitSpain_EN-CA7493953677_UHD.jpg
Being gentle to the giants	bed5284d542e93bdbbb20ab6cb7dd9ab_OHR.WhaleSharkDay_EN-CA7348725715_UHD.jpg
Dune days and desert haze	810d48dd69d133dedf774d660eee8dcb_OHR.DjanetAlgeria_EN-CA7183702479_UHD.jpg
Row your boat gently down the Thames	15118304a874a8445003ea056cdbe261_OHR.ThamesLondon_EN-CA7037142112_UHD.jpg
The Milky Way takes a bow over Elbow River	40556ddaa5277be3b7e1c757e9688da7_OHR.ElbowRiver_EN-CA6581725556_UHD.jpg
Unspoiled beauty for all	b08dc4075e432fbf73d125df410aa184_OHR.AlpineLakes_EN-CA4889089553_UHD.jpg
Birds of a feather hoot together	7b3a45995729fb5aad72e6e9de557764_OHR.DuskyOwls_EN-CA5479353295_UHD.jpg
Lights, camera, TIFF '24!	7cbccf1d85e21fc8938084001fd8d8fe_OHR.TIFF2024_EN-CA6309124110_UHD.jpg
Wood' you believe this view?	6ac6bea601a7cec733427ab9c1af257f_OHR.GlenariffPark_EN-CA9582120244_UHD.jpg
Humming along	5f15bff8a7abff5f3ed63292e5348bd0_OHR.SantaCruzHummer_EN-CA9641643755_UHD.jpg
A global chapter Unlocking minds	299296768444caef6dbd3816caaa140c_OHR.StockholmLibrary_EN-CA2154287662_UHD.jpg
